ls1intum/Artemis
# `Development`: Move lecture video unit features into the lecture micro service

### Checklist
#### General


#### Server
### Motivation and Context


This PR moves functionality from Artemis into the Lecture microservice. It moves the lecture video unit resource and its tests.

### Description
The PR doesn't add new features, we just want to make sure the old ones are still working in the new architecture.

### Steps for Testing

Unfortunately, it is still not possible to test this PR on a test server. It should be tested locally. 
n order to do it, you need to run:
- JHipster Registry by running `docker-compose -f src/main/docker/jhipster-registry.yml up`
- Gateway application by running `./gradlew :gateway:bootRun`
- Lecture application by running `./gradlew :lecture:bootRun --args='--spring.profiles.active=artemis,dev'`
- Artemis as usual or i.e. `./gradlew bootRun --args='--spring.profiles.active=dev,bamboo,bitbucket,artemis,scheduling' -x webapp` to skip the client code build since it's not needed

Prerequisites:
- 1 Instructor
- 1 Student

0. Open [http://localhost:8089/](http://localhost:8089/)
1. Log in to Artemis
2. Navigate to Course Management -> Lectures -> Units
5. Create a new Video Unit
6. Open it, make sure you can access it 
7. Edit the created video unit and make sure the changes are saved
8. Check that you can access it as a student, too

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
